src/heap/sharedMem.ml

```diff
@@ -1156,36 +1159,38 @@ module NewAPI = struct
 
   let file_patterns file = read_addr (get_heap ()) (patterns_addr file)
 
-  let cjs_module_info m = read_addr (get_heap ()) (m + 1)
+  let cjs_module_info m = read_addr (get_heap ()) (addr_offset m 1)
 
-  let cjs_module_exports m = read_addr (get_heap ()) (m + 2)
+  let cjs_module_exports m = read_addr (get_heap ()) (addr_offset m 2)
 
-  let cjs_module_type_exports m = read_addr (get_heap ()) (m + 3)
+  let cjs_module_type_exports m = read_addr (get_heap ()) (addr_offset m 3)
 
-  let es_module_info m = read_addr (get_heap ()) (m + 1)
+  let es_module_info m = read_addr (get_heap ()) (addr_offset m 1)
 
-  let es_module_exports m = read_addr (get_heap ()) (m + 2)
+  let es_module_exports m = read_addr (get_heap ()) (addr_offset m 2)
 
-  let es_module_type_exports m = read_addr (get_heap ()) (m + 3)
+  let es_module_type_exports m = read_addr (get_heap ()) (addr_offset m 3)
 
   (* write *)
 
+  (* Read int64 from given byte offset in the heap. This is not bounds checked. *)
+  external unsafe_write_int64 : heap -> int -> int64 -> unit = "%caml_bigstring_set64u"
+
   (* Write a header at a specified address in the heap. This write is not
    * bounds checked; caller must ensure the given destination has already been
    * allocated. *)
   let unsafe_write_header_at heap dst hd =
-    Array1.unsafe_set heap dst Nativeint.(logor 1n (shift_left (of_int hd) 2))
+    unsafe_write_int64 heap dst Int64.(logor 1L (shift_left (of_int hd) 2))
 
   (* Write an address at a specified address in the heap. This write is not
    * bounds checked; caller must ensure the given destination has already been
    * allocated. *)
-  let unsafe_write_addr_at heap dst addr =
-    Array1.unsafe_set heap dst Nativeint.(shift_left (of_int addr) 3)
+  let unsafe_write_addr_at heap dst addr = unsafe_write_int64 heap dst (Int64.of_int addr)
 
   (* Write a 64-bit hash at a specified address in the heap. This write is not
    * bounds checked; caller must ensure the given destination has already been
    * allocated. *)
-  let unsafe_write_hash_at heap dst hash = Array1.unsafe_set heap dst (Int64.to_nativeint hash)
+  let unsafe_write_hash_at heap dst hash = unsafe_write_int64 heap dst hash
 
```

Comment:
nit: seems could be simplified as an alias

Reply:
You're right! Will change.